lib/facility_factory.rb

```diff
@@ -0,0 +1,29 @@
+class FacilityFactory
+  def create_facility(api_data)
+    facilities = []
+    api_data.each do |data|
+      if data[:dmv_office] != nil
+      name = data[:dmv_office]
+      address = "#{data[:address_li]} #{data[:address__1]} #{data[:city]} #{data[:state]} #{data[:zip]}"
+      phone = data[:phone]
+      #ny
+      elsif data[:office_name] != nil 
+        name = data[:office_name]
+        address = "#{data[:street_address_line_1]}"
+        address += "#{data[:street_address_line_2]} #{data[:city]} #{data[:state]} #{data[:zip]}"
+        phone = data[:public_phone_number]
+        
+      else
+        #Mo
+        data[:address] != nil 
+        name = data[:name]
+        address = "#{data[:address1]} #{data[:city]} #{data[:state]} #{data[:zip]}"
+        phone = data[:phone]
+      end 
+      facility = Facility.new(name: name, address: address, phone: phone)
+      facilities << facility
+    end 
+    facilities
+    
+  end 
+end              
```

Comment:
The code in this class is good, there are just a few spacing and indentation issues. Consider using consistent indentation and spacing to improve readability.

Finally, while this logic is okay in terms of determining which type of Facility to create, I also believe it is quite fragile. I think the best way to determine what type of Facility to create would have been by using the state data like so:

```ruby
def create_facility(api_data) # This method should also be named plurally
  facilities = []

  api_data.each do |data|
    state = data[:state]

    if state == "CO"
      # Parse the CO data
    elsif state == "NY"
      # Parse the NY data
    elsif state == "MO"
      # Parse the MO data
    else
      # Do something else...
    end
  
    facility = Facility.new(name: name, address: address, phone: phone)
    facilities << facility
  end

  facilities
end
```

This small change makes your code more resistant to breaking since we're relying on the value of the state and not some arbitrary key to not be `nil`.

Also, because we are creating new Facility objects in this class, you need to require the Facility class at the top of this file.